app/assets/js/app.js

```diff
@@ -2,17 +2,23 @@ $(function() {
   $('.map-container').each(function() {
     var container = this;
 
-    $.getJSON(window.location.pathname, function(data) {
-      var map = L.map(container);
-      var geoJson = L.geoJson(data.geometry);
+    $.ajax(window.location.pathname, {
+      dataType: 'json',
+      accepts: {
+        json: 'application/vnd.geo+json'
+      },
+      complete: function(data) {
```

Comment:
Use `success` callback instead of `complete`, otherwise this code will run on errors too.